app/assets/javascripts/ransack_ui_jquery/index.js

```diff
@@ -1,2 +1,6 @@
 //= require ransack/predicates
-//= require ransack_ui_jquery/search_form
\ No newline at end of file
+//= require ransack_ui_jquery/search_form
+
+if(jQuery.fn.select2) {
+  window.Select2 = jQuery.fn.select2;
+}
```

Comment:
agreed, we should probably have a fresh branch for version 4, however for now this is just to make the select2 version 4 backwards compatible with select 2 version 3